conans/model/version_range.py

```diff
@@ -187,6 +195,10 @@ def contains(self, version: Version, resolve_prerelease: Optional[bool]):
         return False
 
     def intersection(self, other):
+        # skip calculations if ranges are equal
+        if self.condition_sets == other.condition_sets:
+            return VersionRange(self._expression)
```

Comment:
Is this an optimization? In theory the intersection algorithm should also work without this pre-check?

If that is the case (and it probably also means that the above ``def __eq__(self, other):`` can be dropped, probably it is better to drop it.

Reply:
Removed it